### PR TITLE
Feature/at 6380 As a mission owner, I need to invite portfolio managers

### PR DESCRIPTION
**Pull request by jGonzalez-CCPO**
Create the UI that will allow mission owners to invite portfolio managers.
The UI will include
blue button (used to open modal) Figure 1
modal (where user invites portfolio managers and gives desired permissions) Figure 2
both button and modal are to be 508 Compliant
user is to successfully tab and click the INVITE PORTFOLIO MANAGER button, focus will be on modal Figure 2
In modal, user is to successfully tab and add single or multiple email addresses and set permissions with checkboxes 

FIGURE 1

FIGURE 2 
 
Figma Prototype for Create Portfolio screen – Click Invite Portfolio Manager button to open modal

Acceptance Criteria
“Invite Portfolio Manager” button is present, and opens modal window
Within the “Invite Portfolio Manager” modal window, the following elements are present... 
Portfolio manager description
Email address input field
“Add another portfolio manager” button, with functionality to add additional email address input fields
“Portfolio Permissions” checkboxes
All elements are accessible via keyboard
All elements are accessible via screen reader

### CODE REVIEW

File: src/components/ATATTextField.vue [hunk 1]
```diff
@@ -39,6 +39,7 @@ export default class ATATTextField extends VTextField {
   @Prop({ default: false }) private success!: boolean;
   @Prop({ default: false }) private error!: boolean;
   @Prop({ default: "id_is_missing" }) private id!: string;
+  @Prop( {default: "text input"}) private label: string;
```
Comment (tonysumner-ccpo) [inline src/components/ATATTextField.vue:42]: replace "text input" to "Form Field Label"

File: src/wizard/Step1/components/PermissionsModal.vue [hunk 2]
```diff
@@ -1,11 +1,52 @@
...
-@Component({})
-export default class PermissionsModal extends Vue {}
+@Component({
+  components: { ATATTextField },
+})
+export default class PermissionsModal extends Vue {
+  @Prop() isDialogVisible = true;
+
+  private doCancel(): void {
+    // do something
```
Comment (tonysumner-ccpo) [inline src/wizard/Step1/components/PermissionsModal.vue:44]: I emitted an event form the button nav bar component to the parent Wizard Component.   Easy to do. Am sure you know,  The code is there is you need to see it.

File: src/wizard/Step1/components/PermissionsModal.vue [hunk 3]
```diff
@@ -1,11 +1,52 @@
...
+import ATATTextField from "@/components/ATATTextField.vue";
 
-@Component({})
-export default class PermissionsModal extends Vue {}
+@Component({
+  components: { ATATTextField },
+})
+export default class PermissionsModal extends Vue {
+  @Prop() isDialogVisible = true;
+
```
Comment (tonysumner-ccpo) [inline src/wizard/Step1/components/PermissionsModal.vue:42]: to set the default value for the prop, code it like this. 
`@Prop({ default: true }) private isDialogVisible!: boolean;`
We can set other attributes of the prop within the parenthesis, such as `type` and  `required`, if necessary.

File: src/wizard/Step1/components/PermissionsModal.vue [hunk 4]
```diff
@@ -1,11 +1,181 @@
...
+    // do something
+    // emit action and object
+    console.log("doSave");
+    this.modalAction("save");
+  }
+}
 </script>
+
+<style lang="scss">
+.portfolio-managers-modal-dialog {
```
Comment (tonysumner-ccpo) [inline src/wizard/Step1/components/PermissionsModal.vue:120]: make sure use as many out-of-the-box veutify classes that you can like for padding and display and position and height.

- [Padding and margin](https://vuetifyjs.com/en/styles/spacing/#negative-margin)
- [display helpers](https://vuetifyjs.com/en/styles/display/#accessibility)
- [Flex](https://vuetifyjs.com/en/styles/flex/)
- [Float] (https://vuetifyjs.com/en/styles/float/#overview)

File: src/store/index.ts [hunk 5]
```diff
@@ -3,6 +3,7 @@ import Vuex from "vuex";
 import VuexPersist from "vuex-persist";
 import { Navs } from "../../types/NavItem";
 import { allPortfolios } from "@/store/mocks/portfoliosMockData";
+import { portfoliManagerPermisions } from "./mocks/managers.mock";
```
Comment (tonysumner-ccpo) [inline src/store/index.ts:6]: should be portfoli**o**ManagerPermis**s**ions

File: src/wizard/Step1/components/PermissionsModal.vue [hunk 6]
```diff
@@ -1,11 +1,209 @@
...
+                  body-lg
+                "
+              :ripple="false"
+              @click="modalAction('portfolio-managers-modal-cancel')"
+            >
+              Close
+            </v-btn>
+            <v-btn
+              color="primary"
+              :disabled="isSaveDisenable"
```
Comment (tonysumner-ccpo) [inline src/wizard/Step1/components/PermissionsModal.vue:101]: should be isSaveDisabled

File: src/wizard/Step1/components/PermissionsModal.vue [hunk 7]
```diff
@@ -1,11 +1,209 @@
...
+          </v-card-title>
+          <v-card-text>
+            <label class="form-field-label"> Email address </label>
+            <div class="form-field-group">
+              <div
+                class="form-field-item"
+                v-for="(manager, counter) in listManagers"
+                v-bind:key="counter"
+              >
+                <v-text-field
```
Comment (tonysumner-ccpo) [inline src/wizard/Step1/components/PermissionsModal.vue:29]: Do not use v-text-field.  Use the ATATTextField instead as it has the proper styling.

File: src/wizard/Step1/components/PermissionsModal.vue [hunk 8]
```diff
@@ -1,11 +1,209 @@
...
+                  height="42"
+                  type="email"
+                  :rules="validEmailRules"
+                  :messages="['Messages']"
+                  v-model.lazy="listManagers[counter]"
+                  placeholder="add e-mail"
+                  hide-details="true"
+                  class="invite-portfolio-manager-email"
+                >
+                </v-text-field>
```
Comment (tonysumner-ccpo) [inline src/wizard/Step1/components/PermissionsModal.vue:41]: the email field should look like this...
image

remove trashcan and add 'x' on the outside of it.

File: src/wizard/Step1/components/PermissionsModal.vue (hunk 7 above)
Comment (jGonzalez-CCPO) [reply src/wizard/Step1/components/PermissionsModal.vue:29]: true i almost forgot

**Pull request closed without merging**